client/js/lounge.js

```diff
@@ -427,18 +444,56 @@ $(function() {
 		return false;
 	});
 
+	const contextMenuActions = {
+		close: function(itemData) {
+			$(`.networks .chan[data-target="${itemData}"] .close`).click();
+		},
+		focusChan: function(itemData) {
+			$(`.networks .chan[data-target="${itemData}"]`).click();
+		},
+		whois: function(itemData) {
+			const chan = utils.findCurrentNetworkChan(itemData);
+
+			if (chan.length) {
+				chan.click();
+			}
+
+			socket.emit("input", {
+				target: $("#chat").data("id"),
+				text: "/whois " + itemData,
+			});
+
+			$(`.channel.active .users .user[data-name="${itemData}"]`).click();
+		},
+		query: function(itemData) {
+			const chan = utils.findCurrentNetworkChan(itemData);
+
+			if (chan.length) {
+				chan.click();
+			}
+
+			socket.emit("input", {
+				target: $("#chat").data("id"),
+				text: "/query " + itemData,
+			});
+		},
+		kick: function(itemData) {
+			socket.emit("input", {
+				target: $("#chat").data("id"),
+				text: "/kick " + itemData,
+			});
+		},
+	};
+
+	contextMenuActions.execute = function(name) {
+		return contextMenuActions[name] && contextMenuActions[name].apply(contextMenuActions, [].slice.call(arguments, 1));
```

Comment:
Why the `.apply` and `[].slice.call`?
It seems to me you could add an `itemData` argument to the `execute` function and pass it to `contextMenuActions[name](itemData)`, no?

Reply:
Simply future prooving and OOP oriented. If the handling method would change it would require all objects directly accessing these methods  to also be modified. By abstracting it away further like this we accept any named methods that can be performed on the object, passing along any data that might be used.

Reply:
Could you try this? It might work and is more explicit/simpler (also FP > OOP 😁):

```js
contextMenuActions.execute = (name, ...args) => contextMenuActions[name] && contextMenuActions[name](args);
```

Reply:
Hrm I guess that could work, didn't think of using a rest parameter.

Reply:
Actually, that would be `contextMenuActions[name](...args)` too, rather.